packager/packager.cc

```diff
@@ -2160,6 +2160,100 @@ vector<ast::ParsedFile> Packager::runIncremental(const core::GlobalState &gs, ve
     return files;
 }
 
+void Packager::buildPackageDB(core::GlobalState &gs, WorkerPool &workers, absl::Span<ast::ParsedFile> files) {
+    ENFORCE(!gs.runningUnderAutogen, "Packager pass does not run in autogen");
+
+    Timer timeit(gs.tracer(), "packager");
+    timeit.setTag("mode", "buildPackageDB");
+
+    findPackages(gs, files);
+    setPackageNameOnFiles(gs, files);
+
+    {
+        Timer timeit(gs.tracer(), "packager.buildPackageDB");
+
+        auto taskq = std::make_shared<ConcurrentBoundedQueue<size_t>>(files.size());
+        absl::BlockingCounter barrier(max(workers.size(), 1));
+
+        for (size_t i = 0; i < files.size(); ++i) {
+            taskq->push(i, 1);
+        }
+
+        if (gs.packageDB().enforceLayering()) {
+            computeSCCs(gs);
+        }
+
+        workers.multiplexJob("rewritePackagesAndFiles", [&gs, &files, &barrier, taskq]() {
+            Timer timeit(gs.tracer(), "packager.rewritePackagesAndFilesWorker");
+            size_t idx;
+            for (auto result = taskq->try_pop(idx); !result.done(); result = taskq->try_pop(idx)) {
+                ast::ParsedFile &job = files[idx];
+                if (result.gotItem()) {
```

Comment:
Shouldn't we be checking `gotItem` prior to using `idx`?  Otherwise we might be using garbage -- at least on the first iteration of the loop.